lib/stackprof/report.rb

```diff
@@ -80,65 +83,36 @@ def print_stackcollapse
       end
     end
 
-    def print_flamegraph(f=STDOUT, skip_common=true)
+    def view_flamegraph_in_browser(filepath)
       raise "profile does not include raw samples (add `raw: true` to collecting StackProf.run)" unless raw = data[:raw]
+      profile_b64 = Base64.strict_encode64(JSON.generate(data))
+      filename = File.basename(filepath)
+      js_source = "speedscope.loadFileFromBase64('#{filename}', '#{profile_b64}')"
 
-      stacks = []
-      max_x = 0
-      max_y = 0
-      while len = raw.shift
-        max_y = len if len > max_y
-        stack = raw.slice!(0, len+1)
-        stacks << stack
-        max_x += stack.last
-      end
-
-      f.puts 'flamegraph(['
-      max_y.times do |y|
-        row_prev = nil
-        row_width = 0
-        x = 0
-
-        stacks.each do |stack|
-          weight = stack.last
-          cell = stack[y] unless y == stack.length-1
-
-          if cell.nil?
-            if row_prev
-              flamegraph_row(f, x - row_width, y, row_width, row_prev)
-            end
-
-            row_prev = nil
-            x += weight
-            next
-          end
+      tmpdir = Dir.mktmpdir("stackprof")
 
-          if row_prev.nil?        # start new row with this cell
-            row_width = weight
-            row_prev = cell
-            x += weight
-
-          elsif row_prev == cell  # grow current row along x-axis
-            row_width += weight
-            x += weight
+      tmp_js_path = File.join(tmpdir, "profile.js")
+      File.open(tmp_js_path, "w") do |tmp_js_file|
+        tmp_js_file.write(js_source)
+      end
+      puts "Creating temp file #{tmp_js_path}"
```

Comment:
Pedantic:  Seems like this should be removed, or possibly hidden behind a `--debug`/`--verbose` flag.

Reply:
Yep! Printing the temp file for the HTML file is useful if the browser open fails for some reason, but I agree this path isn't particularly helpful for anything other than my personal debugging while writing this :)

Reply:
Noticed this after I read this, but maybe putting this down as an `else` case for when `which` can't find a program for a opener, so at least the output file is being printed and something shown.